pkg/providers/snow/entry.go

```diff
@@ -0,0 +1,78 @@
+package snow
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	providerValidator "github.com/aws/eks-anywhere/pkg/providers/validator"
+)
+
+type ConfigManagerEntry struct {
+	configManager *cluster.ConfigManager
+	validator     *Validator
+	defaulters    *Defaulters
+}
+
+func NewConfigManagerEntry(aws aws.Clients, writer filewriter.FileWriter) *ConfigManagerEntry {
```

Comment:
I would pass the validator and the defaulters as args here

Reply:
what's the reason for that?